es/resource_elasticsearch_index.go

```diff
@@ -341,13 +407,35 @@ func resourceElasticsearchIndexRead(d *schema.ResourceData, meta interface{}) er
 		}
 	}
 
-	name := index
-	if providedName, ok := settings["provided_name"].(string); ok {
-		name = providedName
+	// Don't override name otherwise it will force a replacement
+	if _, ok := d.GetOk("name"); !ok {
+		name := index
+		if providedName, ok := settings["provided_name"].(string); ok {
+			name = providedName
+		}
+		err := d.Set("name", name)
+		if err != nil {
+			return err
+		}
 	}
-	err := d.Set("name", name)
-	if err != nil {
-		return err
+
+	// If index is managed by ILM or ISM set rollover_alias
```

Comment:
Thanks for handling both styles, much appreciated. Should we throw an error if rollover_alias is set but the index isn't managed?

Reply:
I think an error in case of `rollover_alias` is set but not managed by ILM/ISM is not necessary.